litgpt/model.py

```diff
@@ -317,12 +348,15 @@ def forward(
             sliding_window_bias.masked_fill_(sliding_window_bias.bool(), float("-inf"))
             mask += sliding_window_bias
 
+        # Efficient attention using Flash Attention CUDA kernels.
```

Comment:
Only efficient if `attention_logit_softcapping` is not used

Reply:
Correct. SDPA turns off flash attention for non-default mask.
For that we need to add FlexAttention to the implementation for a such case.

Reply:
Sometimes I see a text, but read it "differently" :)
Yes, it's disabled for soft-capping because there is a custom implementation is used in pytorch. So no efficient kernels with fusing / memory access optimizations.
But the same is true for masks: if the mask is not None, then the efficient kernel is also not used, since it doesn't "know" what to do with it.

Both cases can be solved with flex-attention. But it's a fight for another day.

Updated the comment.